x-pack/plugins/lens/public/editor_frame_service/editor_frame/expression_helpers.ts

```diff
@@ -95,6 +189,21 @@ export function buildExpression({
     searchSessionId
   );
 
+  // check if any of the datasource expressions should be loaded from cache
+  for (const layerId in datasourceExpressionsByLayers) {
+    if (datasourceExpressionsByLayers[layerId]) {
+      const expr = toExpression(datasourceExpressionsByLayers[layerId]);
+
+      const cacheItem = exprCache.get(layerId);
+      if (cacheItem && cacheItem.expression === expr && cacheItem.value && canUseCache) {
```

Comment:
if we have an item with value in cache and the expression didnt change and we are allowed to reuse the cache ... then lets update the data fetching function to `var`